query-grammar/src/user_input_ast.rs

```diff
@@ -16,6 +16,9 @@ pub enum UserInputLeaf {
         field: Option<String>,
         elements: Vec<String>,
     },
+    Exists {
+        field: Option<String>,
```

Comment:
I don't understand the reason for the `Option`  here.
 Can you add a comment explaining the meaning of `Exists { field: None }` and when it can be built?

Reply:
it is just an intermediary state that should never make it out to the world. I've changed so it's always a String, which can temporarily be empty